src/api/services/health_check.py

```diff
@@ -0,0 +1,86 @@
+import datetime
+import os
+
+from sqlalchemy.exc import SQLAlchemyError
+from telegram.ext import Application
+
+from src.api.constants import DATE_TIME_FORMAT, DOCKER_WORKDIR
+from src.api.schemas import BotStatus, CommitStatus, DBStatus
+from src.core.db.repository import TaskRepository
+from src.settings import settings
+
+
+class HealthCheckService:
+    """Сервис для проверки работы бота."""
+
+    def __init__(self, task_repository: TaskRepository, telegram_bot: Application) -> None:
+        self._repository = task_repository
+        self._bot = telegram_bot
+
+    async def check_bot(self) -> BotStatus:
+        try:
+            webhook_info = await self._bot.bot.get_webhook_info()
+        except Exception as exc:
+            bot_status: BotStatus = {"status": False, "error": f"{exc}"}
+            return bot_status
+        if settings.BOT_WEBHOOK_MODE:
+            bot_status: BotStatus = {"status": True, "method": "webhooks", "url": webhook_info.url}
+            return bot_status
+        bot_status: BotStatus = {"status": True, "method": "pulling"}
+        return bot_status
+
+    async def get_last_commit(self) -> CommitStatus:
+        if str(os.getcwd()) == DOCKER_WORKDIR:
+            commit_status: CommitStatus = {
+                "last_commit": settings.LAST_COMMIT,
+                "commit_date": settings.COMMIT_DATE,
+                "git_tags": settings.TAGS,
+            }
+            return commit_status
+        try:
+            from git import Repo
+            from git.exc import InvalidGitRepositoryError
+
+            Repo(os.getcwd())
+        except NameError as exc:
+            commit_status: CommitStatus = {
+                "last_commit": settings.LAST_COMMIT,
+                "commit_date": settings.COMMIT_DATE,
+                "git_tags": settings.TAGS,
+                "commit_error": f"{exc}",
+            }
+            return commit_status
+        try:
+            repo = Repo(os.getcwd())
+        except InvalidGitRepositoryError as exc:
+            commit_status: CommitStatus = {
+                "last_commit": settings.LAST_COMMIT,
+                "commit_date": settings.COMMIT_DATE,
+                "git_tags": settings.TAGS,
+                "commit_error": f"InvalidGitRepositoryError: {exc}",
+            }
+            return commit_status
```

Comment:
Это можно объединить с первым блоком: 
```suggestion
        except (InvalidGitRepositoryError, NameError) as exc:
            commit_status: CommitStatus = {
                "last_commit": settings.LAST_COMMIT,
                "commit_date": settings.COMMIT_DATE,
                "git_tags": settings.TAGS,
                "commit_error": f"{exc.__name___}: {exc}",
            }
            return commit_status
```